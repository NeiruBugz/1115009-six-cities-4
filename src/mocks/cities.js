export const cities = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const citiesWithCoordinates = [
  {coordinates: [], city: `Paris`},
  {coordinates: [], city: `Cologne`},
  {coordinates: [], city: `Brussels`},
  {coordinates: [52.38333, 4.9], city: `Amsterdam`},
  {coordinates: [], city: `Hamburg`},
  {coordinates: [], city: `Dusseldorf`},
];
