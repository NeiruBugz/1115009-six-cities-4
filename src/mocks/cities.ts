export const cities = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const citiesWithCoordinates = [
  { location: { latitude: 0, longitude: 0 }, name: `Paris` },
  { location: { latitude: 0, longitude: 0 }, name: `Cologne` },
  { location: { latitude: 0, longitude: 0 }, name: `Brussels` },
  { location: { latitude: 52.38333, longitude: 4.9 }, name: `Amsterdam` },
  { location: { latitude: 0, longitude: 0 }, name: `Hamburg` },
  { location: { latitude: 0, longitude: 0 }, name: `Dusseldorf` },
];
