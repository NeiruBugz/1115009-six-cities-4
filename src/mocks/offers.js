export const offers = [
  {
    id: 0,
    mark: `Premium`,
    image: `img/apartment-01.jpg`,
    offerImages: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
    ],
    price: 120,
    priceText: `night`,
    rating: 80,
    type: `House`,
    title: `Wood and stone place`,
    ratingValue: `4.8`,
    bedroomsCount: 3,
    adultsCount: 4,
    features: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cable TV`,
    ],
    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    super: false,
    offerDescription: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    coordinates: [52.3909553943508, 4.85309666406198],
    city: `Amsterdam`,
  },
  {
    id: 1,
    image: `img/apartment-01.jpg`,
    offerImages: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    price: 120,
    priceText: `night`,
    rating: 80,
    type: `Apartment`,
    title: `Wood and stone place`,
    ratingValue: `4.8`,
    bedroomsCount: 3,
    adultsCount: 4,
    features: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cable TV`,
    ],
    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    super: true,
    offerDescription: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    coordinates: [52.369553943508, 4.85309666406198],
    city: `Amsterdam`,
  },
  {
    id: 3,
    image: `img/apartment-01.jpg`,
    offerImages: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    price: 120,
    priceText: `night`,
    rating: 80,
    type: `Hotel`,
    title: `Wood and stone place`,
    ratingValue: `4.8`,
    bedroomsCount: 3,
    adultsCount: 4,
    features: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cable TV`,
    ],
    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    super: false,
    offerDescription: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    coordinates: [52.3909553943508, 4.929309666406198],
    city: `Amsterdam`,
  },
  {
    id: 4,
    mark: `Premium`,
    image: `img/apartment-01.jpg`,
    offerImages: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    price: 120,
    priceText: `night`,
    rating: 80,
    type: `Hotel`,
    title: `Wood and stone place`,
    ratingValue: `4.8`,
    bedroomsCount: 3,
    adultsCount: 4,
    features: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cable TV`,
    ],
    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    super: true,
    offerDescription: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    coordinates: [52.3809553943508, 4.939309666406198],
    city: `Amsterdam`,
  },
  {
    id: 5,
    mark: `Premium`,
    image: `img/apartment-01.jpg`,
    offerImages: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    price: 120,
    priceText: `night`,
    rating: 80,
    type: `House`,
    title: `Wood and stone place`,
    ratingValue: `4.8`,
    bedroomsCount: 3,
    adultsCount: 4,
    features: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cable TV`,
    ],
    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    super: false,
    offerDescription: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    coordinates: [52.369553943508, 4.85309666406198],
    city: `Amsterdam`
  },
  {
    id: 6,
    image: `img/apartment-01.jpg`,
    offerImages: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    price: 120,
    priceText: `night`,
    rating: 80,
    type: `Room`,
    title: `Wood and stone place`,
    ratingValue: `4.8`,
    bedroomsCount: 3,
    adultsCount: 4,
    features: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cable TV`,
    ],
    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    super: true,
    offerDescription: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    coordinates: [52.3909553943508, 4.85309666406198],
    city: `Amsterdam`
  }
];
