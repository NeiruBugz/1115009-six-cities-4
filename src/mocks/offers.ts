/* eslint-disable camelcase */
import { Offer } from "../types/offer.types";

const offers: Offer[] = [
  {
    id: 0,
    is_premium: true,
    preview_image: `img/apartment-01.jpg`,
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
    ],
    price: 120,
    rating: 80,
    type: `House`,
    title: `Wood and stone place`,
    bedrooms: 3,
    max_adults: 4,
    goods: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cable TV`,
    ],
    host: {
      id: 0,
      name: `Angelina`,
      avatar_url: `img/avatar-angelina.jpg`,
      is_pro: false,
    },
    description:
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
      An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    city: {
      name: `Amsterdam`,
      location: {
        longitude: 4.85309666406198,
        latitude: 52.3909553943508,
        zoom: 10
      },
    },
    is_favorite: false,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 12,
    },
  },
  {
    id: 1,
    preview_image: `img/apartment-01.jpg`,
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    price: 120,
    rating: 80,
    type: `Apartment`,
    title: `Wood and stone place`,
    bedrooms: 3,
    max_adults: 4,
    goods: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cable TV`,
    ],
    host: {
      id: 1,
      name: `Angelina`,
      avatar_url: `img/avatar-angelina.jpg`,
      is_pro: true,
    },
    description:
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    location: {
      latitude: 52.369553943508, longitude: 4.85309666406198, zoom: 12
    },
    city: { name: `Amsterdam`, location: { longitude: 4.85309666406198, latitude: 52.3909553943508, zoom: 10 } },
    is_premium: false,
    is_favorite: false,
  },
  {
    id: 2,
    preview_image: `img/apartment-01.jpg`,
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    price: 120,
    rating: 80,
    type: `Apartment`,
    title: `Wood and stone place`,
    bedrooms: 3,
    max_adults: 4,
    goods: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cable TV`,
    ],
    host: {
      id: 1,
      name: `Angelina`,
      avatar_url: `img/avatar-angelina.jpg`,
      is_pro: true,
    },
    description:
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    location: {
      latitude: 52.3909553943508, longitude: 4.929309666406198, zoom: 12
    },
    city: { name: `Amsterdam`, location: { longitude: 4.85309666406198, latitude: 52.3909553943508, zoom: 10 } },
    is_premium: false,
    is_favorite: false,
  },
  {
    id: 3,
    preview_image: `img/apartment-01.jpg`,
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    price: 120,
    rating: 80,
    type: `Apartment`,
    title: `Wood and stone place`,
    bedrooms: 3,
    max_adults: 4,
    goods: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cable TV`,
    ],
    host: {
      id: 1,
      name: `Angelina`,
      avatar_url: `img/avatar-angelina.jpg`,
      is_pro: true,
    },
    description:
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    location: {
      latitude: 52.3809553943508, longitude: 4.939309666406198, zoom: 12
    },
    city: { name: `Amsterdam`, location: { longitude: 4.85309666406198, latitude: 52.3909553943508, zoom: 10 } },
    is_premium: false,
    is_favorite: false,
  },
  {
    id: 4,
    preview_image: `img/apartment-01.jpg`,
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    price: 120,
    rating: 80,
    type: `Apartment`,
    title: `Wood and stone place`,
    bedrooms: 3,
    max_adults: 4,
    goods: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cable TV`,
    ],
    host: {
      id: 1,
      name: `Angelina`,
      avatar_url: `img/avatar-angelina.jpg`,
      is_pro: true,
    },
    description:
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    location: {
      latitude: 52.369553943508, longitude: 4.85309666406198, zoom: 12
    },
    city: { name: `Amsterdam`, location: { longitude: 4.85309666406198, latitude: 52.3909553943508, zoom: 10 } },
    is_premium: false,
    is_favorite: false,
  },
  {
    id: 5,
    preview_image: `img/apartment-01.jpg`,
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    price: 100,
    rating: 70,
    type: `Apartment`,
    title: `Wood and stone place`,
    bedrooms: 3,
    max_adults: 4,
    goods: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cable TV`,
    ],
    host: {
      id: 1,
      name: `Angelina`,
      avatar_url: `img/avatar-angelina.jpg`,
      is_pro: true,
    },
    description:
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    location: {
      latitude: 52.3909553943508, longitude: 4.85309666406198, zoom: 12
    },
    city: { name: `Amsterdam`, location: { longitude: 4.85309666406198, latitude: 52.3909553943508, zoom: 10 } },
    is_premium: false,
    is_favorite: false,
  },
  {
    id: 6,
    preview_image: `img/apartment-01.jpg`,
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    price: 140,
    rating: 100,
    type: `Apartment`,
    title: `Wood and stone place`,
    bedrooms: 3,
    max_adults: 4,
    goods: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cable TV`,
    ],
    host: {
      id: 1,
      name: `Angelina`,
      avatar_url: `img/avatar-angelina.jpg`,
      is_pro: true,
    },
    description:
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    location: {
      latitude: 52.369553943508, longitude: 4.85309666406198, zoom: 12
    },
    city: { name: `Amsterdam`, location: { longitude: 4.85309666406198, latitude: 52.3909553943508, zoom: 10 } },
    is_premium: false,
    is_favorite: false,
  },
];
