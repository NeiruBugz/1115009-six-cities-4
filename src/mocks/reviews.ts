import { ReviewType } from "../types/review.types";

export const reviews: ReviewType[] = [
  {
    id: 0,
    rating: 80,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    date: `April 2019`,
    user: {
      avatar_url: `img/avatar-max.jpg`,
      is_pro: false,
      id: 0,
      name: `Max`
    }
  },
  {
    id: 1,
    rating: 20,
    comment: `Terrible!!!!!!!!`,
    date: `April 2019`,
    user: {
      avatar_url: `img/avatar-max.jpg`,
      is_pro: false,
      id: 0,
      name: `Jack`
    }
  },
  {
    id: 2,
    rating: 100,
    comment: `I've got some good times with this place. Lovely`,
    date: `March 2019`,
    user: {
      avatar_url: `img/avatar-max.jpg`,
      is_pro: false,
      id: 0,
      name: `Melany`
    }
  }
];
