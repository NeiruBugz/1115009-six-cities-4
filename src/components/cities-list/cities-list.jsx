import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {setCity, setOffers} from '../../store/reducer';

import {offers as defOffers} from '../../mocks/offers';

const CitiesList = ({selectedCity, cities, onCityChange, onOffersChange}) => {

  const handleCityClick = (city) => {
    const cityCoordinates = cities.filter((cityObject) => cityObject.name === city)[0].coordinates;
    onCityChange({coordinates: cityCoordinates, name: city});
    onOffersChange(defOffers.filter((offer) => offer.city === city));
  };

  const listMarkup = cities.map((city, index) => {
    return (
      <li className="locations__item" key={`city--${index}`}>
        <a className={`locations__item-link tabs__item ${selectedCity === city.name ? `tabs__item--active` : ``}`}
          onClick={() => handleCityClick(city.name)}
        >
          <span>{city.name}</span>
        </a>
      </li>
    );
  });

  return (
    <section className="locations container">
      <ul className="locations__list tabs__list">
        {listMarkup}
      </ul>
    </section>
  );
};

CitiesList.propTypes = {
  selectedCity: PropTypes.string.isRequired,
  cities: PropTypes.arrayOf(
      PropTypes.shape({
        coordinates: PropTypes.arrayOf(PropTypes.number),
        name: PropTypes.string
      })).isRequired,
  onCityChange: PropTypes.func.isRequired,
  onOffersChange: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  offers: state.offers,
});

const mapDispatchToProps = (dispatch) => ({
  onCityChange: (city) => dispatch(setCity(city)),
  onOffersChange: (offers) => dispatch(setOffers(offers)),
});

export {CitiesList};
// export default connect(mapStateToProps, mapDispatchToProps)(CitiesList);
