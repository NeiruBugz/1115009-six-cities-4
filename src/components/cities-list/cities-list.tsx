import * as React from 'react';
import {connect} from 'react-redux';

import {setCity, setOffers} from '../../store/offers/reducer';

import {City} from "../../types/cities.types";
import {Offer} from "../../types/offer.types";

interface CitiesListProps {
  selectedCity: string;
  cities: City[];
  onCityChange: (city: City) => {};
  onOffersChange: (offers: Offer[]) => {};
  offers?: Offer[];
}

const CitiesList: React.FC<CitiesListProps> = (
    {selectedCity, cities, onCityChange, onOffersChange, offers}
) => {

  const handleCityClick = (city: string) => {
    const cityCoordinates = cities
      .filter(({name}) => name === city)[0].location;
    onCityChange({location: cityCoordinates, name: city});
    onOffersChange(offers.filter((offer: Offer) => offer.city.name === city));
  };

  const listMarkup = cities.map((city, index) => {
    return (
      <li className="locations__item" key={`city--${index}`}>
        <a className={`locations__item-link tabs__item ${selectedCity === city.name ? `tabs__item--active` : ``}`}
          onClick={() => handleCityClick(city.name)}
        >
          <span>{city.name}</span>
        </a>
      </li>
    );
  });

  return (
    <section className="locations container">
      <ul className="locations__list tabs__list">
        {listMarkup}
      </ul>
    </section>
  );
};

const mapStateToProps = (state) => ({
  offers: state.offers.serverOffers,
});

const mapDispatchToProps = (dispatch) => ({
  onCityChange: (city) => dispatch(setCity(city)),
  onOffersChange: (offers) => dispatch(setOffers(offers)),
});

export {CitiesList};

export default connect(mapStateToProps, mapDispatchToProps)(CitiesList);


