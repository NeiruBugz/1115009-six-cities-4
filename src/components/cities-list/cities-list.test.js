import React from 'react';
import renderer from 'react-test-renderer';

import {CitiesList} from './cities-list.jsx';
import {citiesWithCoordinates} from '../../mocks/cities';

it(`should render CitiesList correctly`, function () {
  const citiesList = renderer.create(
      <CitiesList
        onCityChange={() => {}}
        onOffersChange={() => {}}
        cities={citiesWithCoordinates}
        selectedCity={citiesWithCoordinates[0].name}
      />
  ).toJSON();

  expect(citiesList).toMatchSnapshot();
});
