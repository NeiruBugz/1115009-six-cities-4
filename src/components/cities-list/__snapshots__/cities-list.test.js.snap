// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CitiesList correctly 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item "
        onClick={[Function]}
      >
        <span />
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item "
        onClick={[Function]}
      >
        <span />
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item "
        onClick={[Function]}
      >
        <span />
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item "
        onClick={[Function]}
      >
        <span />
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item "
        onClick={[Function]}
      >
        <span />
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item "
        onClick={[Function]}
      >
        <span />
      </a>
    </li>
  </ul>
</section>
`;
